<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Perform Massive Local BLAST Searches • metablastr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Perform Massive Local BLAST Searches">
<meta property="og:description" content="This package provides interface functions for users to perform local Basic Local Alignment Search Tool (BLAST) searches. The search results can be efficiently imported into the R session. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">metablastr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drostlab/metablastr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="metablastr">metablastr<a class="anchor" aria-label="anchor" href="#metablastr"></a>
</h1></div>
<div class="section level2">
<h2 id="seamless-integration-of-blast-sequence-searches-in-r">Seamless Integration of <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="external-link">BLAST</a> Sequence Searches in R<a class="anchor" aria-label="anchor" href="#seamless-integration-of-blast-sequence-searches-in-r"></a>
</h2>
<div class="section level3">
<h3 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h3>
<p>With the rapid expansion of available sequences in biological databases, the landscape of modern life science research is being transformed. Currently, around several hundred thousand genomic sequences from a diverse array of species in the tree of life are freely accessible to the public. The R package <a href="https://github.com/ropensci/biomartr" class="external-link">biomartr</a> enables automated access and retrieval of this vast data, paving the way to delve into the rich tapestry of sequence diversity, uncovering new insights into evolvability, variation, and the emergence of diseases.</p>
<p>The <a href="https://github.com/ropensci/biomartr" class="external-link">biomartr</a> package streamlines the retrieval of a massive amount of biological sequence data in a standardized and reproducible manner. Complementing it, the <code>metablastr</code> package is tailored to facilitate large-scale sequence searches, also in a standardized and reproducible approach.</p>
<p>In synergy, <a href="https://github.com/ropensci/biomartr" class="external-link">biomartr</a> and <code>metablastr</code> provide researchers with a comprehensive toolset, allowing them to efficiently gather thousands of biological sequences (genomes, proteomes, annotations, etc.) and conduct extensive sequence comparisons using the gold standard sequence search engine <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="external-link">BLAST</a>. This facilitates the extraction of novel patterns highlighting similarities and divergences among vast sets of species.</p>
<p>It’s worth noting that the go-to instrument for large-scale sequence searches is BLAST (Basic Local Alignment Search Tool). It is purposefully built to identify regions of sequence similarity between a given query and subject sequences or sequence databases.</p>
<p>Building on these advancements, we have recently introduced <a href="https://www.nature.com/articles/s41592-021-01101-x" class="external-link">DIAMOND2</a>, a groundbreaking software solution designed to accelerate <code>BLAST</code> searches by an factor of up to 10,000x. To offer researchers even more flexibility and integration, we provide <a href="https://github.com/drostlab/rdiamond" class="external-link">rdiamond</a>, a dedicated interface package that allows programmatic handling of <a href="https://github.com/bbuchfink/diamond" class="external-link">DIAMOND2</a> sequence searches directly through R. This not only streamlines the sequence search process but also ensures that researchers can access and utilize the power of <a href="https://github.com/bbuchfink/diamond" class="external-link">DIAMOND2</a> within a familiar R environment.</p>
</div>
<div class="section level3">
<h3 id="short-package-description">Short package description<a class="anchor" aria-label="anchor" href="#short-package-description"></a>
</h3>
<p>The <code>metablastr</code> package harnesses the power of <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="external-link">BLAST</a> by providing interface functions between R and the standalone (command line tool) version of this program.</p>
<p>Together, the <code>metablastr</code> package may enable a new level of data-driven genomics research by providing the computational tools and data science standards needed to perform reproducible research at scale.</p>
</div>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p><strong>The <code>metablastr</code> package is still under development and not formally published yet. However, we did develop parts of <code>metablastr</code> for this Methods chapter which you can cite until a <code>metablastr</code> specific manuscript is prepared.</strong></p>
<blockquote>
<p>M Benoit, HG Drost. <a href="https://link.springer.com/protocol/10.1007%2F978-1-0716-1134-0_1" class="external-link"><strong>A Predictive Approach to Infer the Activity and Natural Variation of Retrotransposon Families in Plants</strong></a>. In: Cho J. (eds) Plant Transposable Elements. Methods in Molecular Biology, vol 2250. Humana, New York, NY (2021).</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="install-metablastr">Install <code>metablastr</code>
<a class="anchor" aria-label="anchor" href="#install-metablastr"></a>
</h3>
</div>
<div class="section level3">
<h3 id="for-linux-users">For Linux Users:<a class="anchor" aria-label="anchor" href="#for-linux-users"></a>
</h3>
<p>Please install the <code>libpq-dev</code> library on you linux machine by typing into the terminal:</p>
<pre><code>sudo apt-get install libpq-dev</code></pre>
</div>
<div class="section level3">
<h3 id="for-all-systems-install-metablastr-by-typing">For all systems install <code>metablastr</code> by typing<a class="anchor" aria-label="anchor" href="#for-all-systems-install-metablastr-by-typing"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install BiocManager if required</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install package dependencies</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Biostrings"</span>,</span>
<span>    <span class="st">"GenomicFeatures"</span>,</span>
<span>    <span class="st">"GenomicRanges"</span>,</span>
<span>    <span class="st">"Rsamtools"</span>,</span>
<span>    <span class="st">"IRanges"</span>,</span>
<span>    <span class="st">"rtracklayer"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install.packages("devtools")</span></span>
<span><span class="co"># install the current version of metablastr on your system</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"drostlab/metablastr"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><strong>Please follow the <a href="https://drostlab.github.io/metablastr/articles/installation.html" class="external-link">Installation Vignette</a> to install all standalone sequence search tools.</strong></p>
</div>
<div class="section level3">
<h3 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drostlab/metablastr" class="external-link">metablastr</a></span><span class="op">)</span></span>
<span><span class="co"># run blastn (nucleotide to nucleotide search) between example query and subject sequences</span></span>
<span><span class="va">blast_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/blast_nucleotide_to_nucleotide.html">blast_nucleotide_to_nucleotide</a></span><span class="op">(</span></span>
<span>                 query   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'seqs/qry_nn.fa'</span>, package <span class="op">=</span> <span class="st">'metablastr'</span><span class="op">)</span>,</span>
<span>                 subject <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'seqs/sbj_nn.fa'</span>, package <span class="op">=</span> <span class="st">'metablastr'</span><span class="op">)</span>,</span>
<span>                 output.path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                 db.import  <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>                 </span>
<span><span class="co"># look at BLAST results</span></span>
<span><span class="va">blast_test</span></span></code></pre></div>
<pre><code>   query_id     subject_id perc_identity num_ident_match… alig_length mismatches
   &lt;chr&gt;        &lt;chr&gt;              &lt;dbl&gt;            &lt;int&gt;       &lt;int&gt;      &lt;int&gt;
 1 333554|PACi… AT1G01010…          84.2              640         760         63
 2 333554|PACi… AT1G01010…          84.0              536         638         90
 3 333554|PACi… AT1G01010…          78.6               44          56         12
 4 470181|PACi… AT1G01020…          94.7              699         738         39
 5 470180|PACi… AT1G01030…          95.2             1025        1077         40
 6 333551|PACi… AT1G01040…          96.0             3627        3779        125
 7 333551|PACi… AT1G01040…          95.5             1860        1948         82
 8 909874|PACi… AT1G01050…          96.6              617         639         22
 9 470177|PACi… AT1G01060…          92.8             1804        1944        110
10 918864|PACi… AT1G01070…          95.3             1046        1098         40
# ... with 13 more rows, and 15 more variables: gap_openings &lt;int&gt;,
#   n_gaps &lt;int&gt;, pos_match &lt;int&gt;, ppos &lt;dbl&gt;, q_start &lt;int&gt;, q_end &lt;int&gt;,
#   q_len &lt;int&gt;, qcov &lt;int&gt;, qcovhsp &lt;int&gt;, s_start &lt;int&gt;, s_end &lt;dbl&gt;,
#   s_len &lt;dbl&gt;, evalue &lt;dbl&gt;, bit_score &lt;chr&gt;, score_raw &lt;int&gt;</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="discussions-and-bug-reports">Discussions and Bug Reports<a class="anchor" aria-label="anchor" href="#discussions-and-bug-reports"></a>
</h2>
<p>I would be very happy to learn more about potential improvements of the concepts and functions provided in this package.</p>
<p>Furthermore, in case you find some bugs, need additional (more flexible) functionality of parts of this package, or want to contribute to this project please let me know:</p>
<p><a href="https://github.com/drostlab/metablastr/issues" class="external-link uri">https://github.com/drostlab/metablastr/issues</a></p>
</div>
<div class="section level2">
<h2 id="interfaces-implemented-in-metablastr">Interfaces implemented in <code>metablastr</code>:<a class="anchor" aria-label="anchor" href="#interfaces-implemented-in-metablastr"></a>
</h2>
<div class="section level3">
<h3 id="perform-blast-searches">Perform BLAST searches<a class="anchor" aria-label="anchor" href="#perform-blast-searches"></a>
</h3>
<ul>
<li>
<code><a href="reference/blast_protein_to_protein.html">blast_protein_to_protein()</a></code>: Perform Protein to Protein BLAST Searches (BLASTP)</li>
<li>
<code><a href="reference/blast_nucleotide_to_nucleotide.html">blast_nucleotide_to_nucleotide()</a></code>: Perform Nucleotide to Nucleotide BLAST Searches (BLASTN)</li>
<li>
<code><a href="reference/blast_nucleotide_to_protein.html">blast_nucleotide_to_protein()</a></code>: Perform Nucleotide to Protein BLAST Searches (BLASTX)</li>
<li>
<code><a href="reference/blast_protein_to_nucleotide.html">blast_protein_to_nucleotide()</a></code>: Perform Protein to Nucleotide BLAST Searches (TBLASTN)</li>
<li>
<code><a href="reference/blast_best_hit.html">blast_best_hit()</a></code>: Retrieve only the best BLAST hit for each query</li>
<li>
<code><a href="reference/blast_best_reciprocal_hit.html">blast_best_reciprocal_hit()</a></code>: Retrieve only the best reciprocal BLAST hit for each query</li>
<li>
<code>blast_rpsblast</code>: Perform Reverse PSI-BLAST searches (rpsblast)</li>
<li>
<code><a href="reference/read_blast.html">read_blast()</a></code>: Import BLAST output into R session (in memory) or via <code>PostgresSQL</code> database connection.</li>
</ul>
</div>
<div class="section level3">
<h3 id="blast-against-common-ncbi-databases">BLAST against common NCBI databases<a class="anchor" aria-label="anchor" href="#blast-against-common-ncbi-databases"></a>
</h3>
<ul>
<li>
<code><a href="reference/blast_protein_to_nr_database.html">blast_protein_to_nr_database()</a></code>: Perform Protein to Protein BLAST Searches against the <code>NCBI non-redundant database</code>
</li>
<li>
<code>blast_nt()</code>: Perform Nucleotide to Nucleotide BLAST Searches against the <code>NCBI non-redundant database</code>
</li>
<li>
<code>blast_est()</code>: Perform Nucleotide to Nucleotide BLAST Searches against the <code>NCBI expressed sequence tags database</code>
</li>
<li>
<code>blast_pdb_protein()</code>:</li>
<li>
<code>blast_pdb_nucleotide()</code>:</li>
<li>
<code>blast_swissprot()</code>:</li>
<li>
<code>blast_delta()</code>:</li>
<li>
<code>blast_refseq_rna()</code>:</li>
<li>
<code>blast_refseq_gene()</code>:</li>
<li>
<code>blast_refseq_protein()</code>:</li>
</ul>
</div>
<div class="section level3">
<h3 id="blast-against-a-set-of-organisms">BLAST against a set of organisms<a class="anchor" aria-label="anchor" href="#blast-against-a-set-of-organisms"></a>
</h3>
<ul>
<li>
<code><a href="reference/blast_nucleotide_to_genomes.html">blast_nucleotide_to_genomes()</a></code>: Perfrom BLAST Searches Against a Set of Genomes</li>
<li>
<code><a href="reference/blast_protein_to_proteomes.html">blast_protein_to_proteomes()</a></code>: Perfrom BLAST Searches Against a Set of Proteomes</li>
<li>
<code><a href="reference/detect_homologs_cds_to_cds.html">detect_homologs_cds_to_cds()</a></code>: Perform CDS to CDS BLAST Searches against a set of CDS files</li>
<li>
<code><a href="reference/detect_homologs_proteome_to_proteome.html">detect_homologs_proteome_to_proteome()</a></code>: Perform Proteome to Proteome BLAST Searches against a set of Proteomes</li>
<li>
<code><a href="reference/extract_hit_seqs_from_genomes.html">extract_hit_seqs_from_genomes()</a></code>: Extract sequences of BLAST hits in respective genomes and store it as ‘fasta’ file(s)</li>
<li>
<code><a href="reference/extract_random_seqs_from_genome.html">extract_random_seqs_from_genome()</a></code>: Extract random loci from a genome of interest</li>
<li>
<code><a href="reference/sample_chromosome_intervals.html">sample_chromosome_intervals()</a></code>: Helper function to sample random intervals of length ‘interval_width’ from chromosomes</li>
</ul>
</div>
<div class="section level3">
<h3 id="analyze-blast-report">Analyze BLAST Report<a class="anchor" aria-label="anchor" href="#analyze-blast-report"></a>
</h3>
<ul>
<li>
<code>filter_blast_</code>:</li>
</ul>
</div>
<div class="section level3">
<h3 id="navigation-functions">Navigation functions<a class="anchor" aria-label="anchor" href="#navigation-functions"></a>
</h3>
<ul>
<li>
<code><a href="reference/list_outformats.html">list_outformats()</a></code>: List available BLAST output formats</li>
</ul>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/drostlab/metablastr/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/drostlab/metablastr/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2" class="external-link">GPL-2</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing metablastr</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Hajk-Georg Drost <br><small class="roles"> Author, contributor, maintainer </small> <a href="https://orcid.org/http://orcid.org/0000-0002-1567-306X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Anna Gogleva <br><small class="roles"> Author, contributor </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Hajk-Georg Drost, Anna Gogleva.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
