<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Perfrom BLAST Searches Against a Set of Genomes — blast_nucleotide_to_genomes • metablastr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Perfrom BLAST Searches Against a Set of Genomes — blast_nucleotide_to_genomes"><meta property="og:description" content="This function takes a fasta file containing query sequences
as input and performs BLAST searches of these query sequences against
a set of reference genomes to retrieve corresponding hits in diverse genomes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metablastr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/metablastr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perfrom BLAST Searches Against a Set of Genomes</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/metablastr/blob/HEAD/R/blast_nucleotide_to_genomes.R" class="external-link"><code>R/blast_nucleotide_to_genomes.R</code></a></small>
    <div class="hidden name"><code>blast_nucleotide_to_genomes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes a fasta file containing query sequences
as input and performs BLAST searches of these query sequences against
a set of reference genomes to retrieve corresponding hits in diverse genomes.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">blast_nucleotide_to_genomes</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  <span class="va">subject_genomes</span>,</span>
<span>  blast_type <span class="op">=</span> <span class="st">"blastn"</span>,</span>
<span>  blast_output_path <span class="op">=</span> <span class="st">"blast_output"</span>,</span>
<span>  min_alig_length <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  evalue <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>  max.target.seqs <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>  update <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>path to input file in fasta format.</p></dd>


<dt>subject_genomes</dt>
<dd><p>a vector containing file paths to the reference genomes that shall be queried (e.g. file paths returned by <code><a href="https://docs.ropensci.org/biomartr/reference/meta.retrieval.html" class="external-link">meta.retrieval</a></code>).</p></dd>


<dt>blast_type</dt>
<dd><p>specification of the BLAST type shall be used to perform BLAST searches between query and reference.
Available options are:</p><ul><li><p><code>blast_type = "blastn"</code> for  nucleotide to nucleotide BLAST searches (see <code><a href="blast_nucleotide_to_nucleotide.html">blast_nucleotide_to_nucleotide</a></code> for details)</p></li>
<li><p><code>blast_type = "tblastn"</code> for protein to nucleotide BLAST searches (see <code><a href="blast_protein_to_nucleotide.html">blast_protein_to_nucleotide</a></code> for details)</p></li>
</ul></dd>


<dt>blast_output_path</dt>
<dd><p>a path to a folder that will be created to store BLAST output tables for each individual query-genome search.</p></dd>


<dt>min_alig_length</dt>
<dd><p>minimum alignment length that shall be retained in the result dataset. All hit alignments with smaller
hit alignment length will be removed automatically.</p></dd>


<dt>evalue</dt>
<dd><p>minimum expectation value (E) threshold for retaining hits (default: evalue = 0.00001).</p></dd>


<dt>max.target.seqs</dt>
<dd><p>maximum number of aligned sequences that shall be retained. Please be aware that <code>max.target.seqs</code> selects best hits based on the database entry and not by the best e-value. See details here: https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty833/5106166 .</p></dd>


<dt>update</dt>
<dd><p>a logical value indicating whether or not pre-computed BLAST tables should be removed and re-computed (<code>update = TRUE</code>)
or imported from existing file (<code>update = FALSE</code>) (Default).</p></dd>


<dt>...</dt>
<dd><p>additional arguments passed to <code><a href="blast_nucleotide_to_nucleotide.html">blast_nucleotide_to_nucleotide</a></code> or <code><a href="blast_protein_to_nucleotide.html">blast_protein_to_nucleotide</a></code>.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>The <code>blast_nucleotide_to_genomes</code> function enables users to BLAST specific query sequences against a set of reference genomes
and retrieve the corresponding BLAST output.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Anna Gogleva.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

