<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Perform Nucleotide to Protein BLAST Searches (BLASTX) — blast_nucleotide_to_protein • metablastr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Perform Nucleotide to Protein BLAST Searches (BLASTX) — blast_nucleotide_to_protein"><meta property="og:description" content="Run nucleotide to protein BLAST of reference sequences
against a blast-able database or fasta file. Internally BLAST translates the 
nucleotide sequence into a protein sequence and then searches for hits."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metablastr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/metablastr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform Nucleotide to Protein BLAST Searches (BLASTX)</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/metablastr/blob/HEAD/R/blast_nucleotide_to_protein.R" class="external-link"><code>R/blast_nucleotide_to_protein.R</code></a></small>
    <div class="hidden name"><code>blast_nucleotide_to_protein.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run nucleotide to protein BLAST of reference sequences
against a blast-able database or fasta file. Internally BLAST translates the 
nucleotide sequence into a protein sequence and then searches for hits.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">blast_nucleotide_to_protein</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  <span class="va">subject</span>,</span>
<span>  strand <span class="op">=</span> <span class="st">"both"</span>,</span>
<span>  output.path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  is.subject.db <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  task <span class="op">=</span> <span class="st">"blastx"</span>,</span>
<span>  db.import <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  postgres.user <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  evalue <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  out.format <span class="op">=</span> <span class="st">"csv"</span>,</span>
<span>  cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  max.target.seqs <span class="op">=</span> <span class="fl">10000L</span>,</span>
<span>  db.soft.mask <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  db.hard.mask <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  blast.path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>path to input file in fasta format.</p></dd>


<dt>subject</dt>
<dd><p>path to subject file in fasta format or blast-able database.</p></dd>


<dt>strand</dt>
<dd><p>Query DNA strand(s) to search against database/subject.
Options are:</p><ul><li><p><code>strand = "both"</code> : query against both DNA strands.</p></li>
<li><p><code>strand = "minus"</code> : query against minus DNA strand.</p></li>
<li><p><code>strand = "plus"</code> : query against plus DNA strand.</p></li>
</ul></dd>


<dt>output.path</dt>
<dd><p>path to folder at which BLAST output table shall be stored. 
Default is <code>output.path = NULL</code> (hence <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code> is used).</p></dd>


<dt>is.subject.db</dt>
<dd><p>logical specifying whether or not the <code>subject</code> file is a file in fasta format (<code>is.subject.db = FALSE</code>; default)
or a <code>fasta</code> file that was previously converted into a blast-able database using <code>makeblastdb</code> (<code>is.subject.db = TRUE</code>).</p></dd>


<dt>task</dt>
<dd><p>nucleotide search task option. Options are:</p><ul><li><p><code>task = "blastx"</code> : Standard nucleotide-protein comparisons (default).</p></li>
<li><p><code>task = "blastx-fast"</code> : Optimized nucleotide-protein comparisons.</p></li>
</ul></dd>


<dt>db.import</dt>
<dd><p>shall the BLAST output be stored in a PostgresSQL database and shall a connection be established to this database? Default is <code>db.import = FALSE</code>.
In case users wish to to only generate a BLAST output file without importing it to the current R session they can specify <code>db.import = NULL</code>.</p></dd>


<dt>postgres.user</dt>
<dd><p>when <code>db.import = TRUE</code> and <code>out.format = "postgres"</code> is selected, the BLAST output is imported and stored in a 
PostgresSQL database. In that case, users need to have PostgresSQL installed and initialized on their system. 
Please consult the Installation Vignette for details.</p></dd>


<dt>evalue</dt>
<dd><p>Expectation value (E) threshold for saving hits (default: <code>evalue = 0.001</code>).</p></dd>


<dt>out.format</dt>
<dd><p>a character string specifying the format of the file in which the BLAST results shall be stored.
Available options are:</p><ul><li><p><code>out.format = "pair"</code> : Pairwise</p></li>
<li><p><code>out.format = "qa.ident"</code> : Query-anchored showing identities</p></li>
<li><p><code>out.format = "qa.nonident"</code> : Query-anchored no identities</p></li>
<li><p><code>out.format = "fq.ident"</code> : Flat query-anchored showing identities</p></li>
<li><p><code>out.format = "fq.nonident"</code> : Flat query-anchored no identities</p></li>
<li><p><code>out.format = "xml"</code> : XML</p></li>
<li><p><code>out.format = "tab"</code> : Tabular separated file</p></li>
<li><p><code>out.format = "tab.comment"</code> : Tabular separated file with comment lines</p></li>
<li><p><code>out.format = "ASN.1.text"</code> : Seqalign (Text ASN.1)</p></li>
<li><p><code>out.format = "ASN.1.binary"</code> : Seqalign (Binary ASN.1)</p></li>
<li><p><code>out.format = "csv"</code> : Comma-separated values</p></li>
<li><p><code>out.format = "ASN.1"</code> : BLAST archive (ASN.1)</p></li>
<li><p><code>out.format = "json.seq.aln"</code> : Seqalign (JSON)</p></li>
<li><p><code>out.format = "json.blast.multi"</code> : Multiple-file BLAST JSON</p></li>
<li><p><code>out.format = "xml2.blast.multi"</code> : Multiple-file BLAST XML2</p></li>
<li><p><code>out.format = "json.blast.single"</code> : Single-file BLAST JSON</p></li>
<li><p><code>out.format = "xml2.blast.single"</code> : Single-file BLAST XML2</p></li>
<li><p><code>out.format = "SAM"</code> : Sequence Alignment/Map (SAM)</p></li>
<li><p><code>out.format = "report"</code> : Organism Report</p></li>
</ul></dd>


<dt>cores</dt>
<dd><p>number of cores for parallel BLAST searches.</p></dd>


<dt>max.target.seqs</dt>
<dd><p>maximum number of aligned sequences that shall be retained. Please be aware that <code>max.target.seqs</code> selects best hits based on the database entry and not by the best e-value. See details here: https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty833/5106166 .</p></dd>


<dt>db.soft.mask</dt>
<dd><p>shall low complexity regions be soft masked? Default is <code>db.soft.mask = FALSE</code>.</p></dd>


<dt>db.hard.mask</dt>
<dd><p>shall low complexity regions be hard masked? Default is <code>db.hard.mask = FALSE</code>.</p></dd>


<dt>blast.path</dt>
<dd><p>path to BLAST executables.</p></dd>

</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="blast_nucleotide_to_nucleotide.html">blast_nucleotide_to_nucleotide</a></code>, <code><a href="blast_protein_to_protein.html">blast_protein_to_protein</a></code>, 
<code><a href="blast_protein_to_nucleotide.html">blast_protein_to_nucleotide</a></code>, <code><a href="blast_best_hit.html">blast_best_hit</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">blast_test</span> <span class="op">&lt;-</span> <span class="fu">blast_nucleotide_to_protein</span><span class="op">(</span></span></span>
<span class="r-in"><span>                 query   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'seqs/qry_nn.fa'</span>, package <span class="op">=</span> <span class="st">'metablastr'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                 subject <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'seqs/sbj_aa.fa'</span>, package <span class="op">=</span> <span class="st">'metablastr'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                 output.path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                 db.import  <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span>                 </span></span>
<span class="r-in"><span> <span class="co"># look at results</span></span></span>
<span class="r-in"><span> <span class="va">blast_test</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Anna Gogleva.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

