<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ metablastr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metablastr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/metablastr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="blast2bed.html">blast2bed()</a></code> </p>
        </td>
        <td><p>Export BLAST output table to BED file</p></td>
      </tr><tr><td>
          <p><code><a href="blast_best_hit.html">blast_best_hit()</a></code> </p>
        </td>
        <td><p>Retrieve only the best BLAST hit for each query</p></td>
      </tr><tr><td>
          <p><code><a href="blast_best_reciprocal_hit.html">blast_best_reciprocal_hit()</a></code> </p>
        </td>
        <td><p>Retrieve only the best reciprocal BLAST hit for each query</p></td>
      </tr><tr><td>
          <p><code><a href="blast_genomes.html">blast_genomes()</a></code> </p>
        </td>
        <td><p>Perfrom BLAST Searches Against a Set of Genomes</p></td>
      </tr><tr><td>
          <p><code><a href="blast_nucleotide_to_genomes.html">blast_nucleotide_to_genomes()</a></code> </p>
        </td>
        <td><p>Perfrom BLAST Searches Against a Set of Genomes</p></td>
      </tr><tr><td>
          <p><code><a href="blast_nucleotide_to_nucleotide.html">blast_nucleotide_to_nucleotide()</a></code> </p>
        </td>
        <td><p>Perform Nucleotide to Nucleotide BLAST Searches (BLASTN)</p></td>
      </tr><tr><td>
          <p><code><a href="blast_nucleotide_to_protein.html">blast_nucleotide_to_protein()</a></code> </p>
        </td>
        <td><p>Perform Nucleotide to Protein BLAST Searches (BLASTX)</p></td>
      </tr><tr><td>
          <p><code><a href="blast_protein_to_nr_database.html">blast_protein_to_nr_database()</a></code> </p>
        </td>
        <td><p>Perform Protein to Protein BLAST Searches Against the NCBI NR Database</p></td>
      </tr><tr><td>
          <p><code><a href="blast_protein_to_nucleotide.html">blast_protein_to_nucleotide()</a></code> </p>
        </td>
        <td><p>Perform Protein to Nucleotide BLAST Searches (TBLASTN)</p></td>
      </tr><tr><td>
          <p><code><a href="blast_protein_to_protein.html">blast_protein_to_protein()</a></code> </p>
        </td>
        <td><p>Perform Protein to Protein BLAST Searches (BLASTP)</p></td>
      </tr><tr><td>
          <p><code><a href="blast_protein_to_proteomes.html">blast_protein_to_proteomes()</a></code> </p>
        </td>
        <td><p>Perfrom BLAST Searches Against a Set of Proteomes</p></td>
      </tr><tr><td>
          <p><code><a href="blast_repbase.html">blast_repbase()</a></code> </p>
        </td>
        <td><p>Query the RepBase to annotate putative LTRs</p></td>
      </tr><tr><td>
          <p><code><a href="blast_rpsblast.html">blast_rpsblast()</a></code> </p>
        </td>
        <td><p>Perform Reverse PSI-BLAST searches (rpsblast)</p></td>
      </tr><tr><td>
          <p><code><a href="detect_homologs_cds_to_cds.html">detect_homologs_cds_to_cds()</a></code> </p>
        </td>
        <td><p>Perform CDS to CDS BLAST Searches against a set of CDS files</p></td>
      </tr><tr><td>
          <p><code><a href="detect_homologs_proteome_to_proteome.html">detect_homologs_proteome_to_proteome()</a></code> </p>
        </td>
        <td><p>Perform Proteome to Proteome BLAST Searches against a set of Proteomes</p></td>
      </tr><tr><td>
          <p><code><a href="extract_hit_seqs_from_genomes.html">extract_hit_seqs_from_genomes()</a></code> </p>
        </td>
        <td><p>Extract sequences of BLAST hits in respective genomes and store it as <code>fasta</code> file(s)</p></td>
      </tr><tr><td>
          <p><code><a href="extract_promotor_seqs_from_genome.html">extract_promotor_seqs_from_genome()</a></code> </p>
        </td>
        <td><p>Retrieve all upstream promotor sequences from a genome</p></td>
      </tr><tr><td>
          <p><code><a href="extract_random_seqs_from_genome.html">extract_random_seqs_from_genome()</a></code> </p>
        </td>
        <td><p>Extract random loci from a genome of interest</p></td>
      </tr><tr><td>
          <p><code><a href="extract_random_seqs_from_multiple_genomes.html">extract_random_seqs_from_multiple_genomes()</a></code> </p>
        </td>
        <td><p>Extract random loci from a set of genomes</p></td>
      </tr><tr><td>
          <p><code><a href="extract_upstream_promotor_seqs.html">extract_upstream_promotor_seqs()</a></code> </p>
        </td>
        <td><p>Retrieve all upstream promotor sequences from a genome</p></td>
      </tr><tr><td>
          <p><code><a href="file_contains_aa.html">file_contains_aa()</a></code> </p>
        </td>
        <td><p>Helper function to check whether input file contains amino acids as required</p></td>
      </tr><tr><td>
          <p><code><a href="file_contains_dna.html">file_contains_dna()</a></code> </p>
        </td>
        <td><p>Helper function to check whether input file contains dna as required</p></td>
      </tr><tr><td>
          <p><code><a href="filter_best_hits.html">filter_best_hits()</a></code> </p>
        </td>
        <td><p>Retrieve the best hits across species from a BLAST table</p></td>
      </tr><tr><td>
          <p><code><a href="filter_homologs_core_set.html">filter_homologs_core_set()</a></code> </p>
        </td>
        <td><p>Retrieve a core set of best hits that is shared across species</p></td>
      </tr><tr><td>
          <p><code><a href="filter_homologs_core_set_n_species.html">filter_homologs_core_set_n_species()</a></code> </p>
        </td>
        <td><p>Retrieve a core set of best hits that is shared across at least N species</p></td>
      </tr><tr><td>
          <p><code><a href="gg_blast_hits.html">gg_blast_hits()</a></code> </p>
        </td>
        <td><p>Plot <code>blast_nucleotide_to_genomes</code> Result</p></td>
      </tr><tr><td>
          <p><code><a href="gg_blast_hits_summary.html">gg_blast_hits_summary()</a></code> </p>
        </td>
        <td><p>Plot <code>blast_nucleotide_to_genomes</code> Result</p></td>
      </tr><tr><td>
          <p><code><a href="gg_hits_boxplot.html">gg_hits_boxplot()</a></code> </p>
        </td>
        <td><p>Boxplot of a <code>blast_tbl</code> generated with <code>detect_homologs_genome_to_genome</code> or <code>detect_homologs_proteome_to_proteome</code></p></td>
      </tr><tr><td>
          <p><code><a href="gg_homologs_one_to_one.html">gg_homologs_one_to_one()</a></code> </p>
        </td>
        <td><p>Plot number of BLAST hits per species as barplot from a <code>tibble</code> generated with <code>detect_homologs_genome_to_genome</code> or <code>detect_homologs_proteome_to_proteome</code></p></td>
      </tr><tr><td>
          <p><code><a href="gg_pairwise_orthologs_line.html">gg_pairwise_orthologs_line()</a></code> </p>
        </td>
        <td><p>Plot number of pairwise orthologs for multiple species comparisons</p></td>
      </tr><tr><td>
          <p><code><a href="gg_species_dnds_blast_tbl.html">gg_species_dnds_blast_tbl()</a></code> </p>
        </td>
        <td><p>Plot pairwise dN, dS, or dN/dS distributions of orthologous genes</p></td>
      </tr><tr><td>
          <p><code><a href="gg_species_feature_blast_tbl.html">gg_species_feature_blast_tbl()</a></code> </p>
        </td>
        <td><p>Plot pairwise feature distributions of orthologous genes</p></td>
      </tr><tr><td>
          <p><code><a href="gg_species_promotor_dist_blast_tbl.html">gg_species_promotor_dist_blast_tbl()</a></code> </p>
        </td>
        <td><p>Plot pairwise promotor sequence distance distributions of orthologous genes</p></td>
      </tr><tr><td>
          <p><code><a href="list_outformats.html">list_outformats()</a></code> </p>
        </td>
        <td><p>List available BLAST output formats</p></td>
      </tr><tr><td>
          <p><code><a href="motif_compare.html">motif_compare()</a></code> </p>
        </td>
        <td><p>Count the number of motifs in a set of non-random versus random sequences</p></td>
      </tr><tr><td>
          <p><code><a href="motif_compare_multi.html">motif_compare_multi()</a></code> </p>
        </td>
        <td><p>Count the number of motifs in a set of non-random versus randomly sampled genomic sequences for multiple species</p></td>
      </tr><tr><td>
          <p><code><a href="motif_compare_multi_promotor_seqs.html">motif_compare_multi_promotor_seqs()</a></code> </p>
        </td>
        <td><p>Count the number of motifs in a set of non-random versus randomly drawn gene promotor sequences for multiple species</p></td>
      </tr><tr><td>
          <p><code><a href="motif_count.html">motif_count()</a></code> </p>
        </td>
        <td><p>Count the number of motifs in a set of sequences</p></td>
      </tr><tr><td>
          <p><code><a href="motif_count_matrix.html">motif_count_matrix()</a></code> </p>
        </td>
        <td><p>Count the number of motifs in a set of sequences and generate a motif count matrix</p></td>
      </tr><tr><td>
          <p><code><a href="motif_enrichment.html">motif_enrichment()</a></code> </p>
        </td>
        <td><p>Test motif enrichment in a set of non-random versus random sequences</p></td>
      </tr><tr><td>
          <p><code><a href="motif_enrichment_multi.html">motif_enrichment_multi()</a></code> </p>
        </td>
        <td><p>Statistical assessment of motif enrichments in a set of non-random versus randomly sampled genomic sequences for multiple species</p></td>
      </tr><tr><td>
          <p><code><a href="motif_enrichment_multi_promotor_seqs.html">motif_enrichment_multi_promotor_seqs()</a></code> </p>
        </td>
        <td><p>Statistical assessment of motif enrichments in a set of non-random versus randomly sampled gene promotor sequences for multiple species</p></td>
      </tr><tr><td>
          <p><code><a href="read_blast.html">read_blast()</a></code> </p>
        </td>
        <td><p>Import BLAST output file</p></td>
      </tr><tr><td>
          <p><code><a href="read_blast_databases.html">read_blast_databases()</a></code> </p>
        </td>
        <td><p>Import BLAST output file</p></td>
      </tr><tr><td>
          <p><code><a href="sample_chromosome_intervals.html">sample_chromosome_intervals()</a></code> </p>
        </td>
        <td><p>Helper function to sample random intervals of length <code>interval_width</code> from chromosomes</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Anna Gogleva.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

