<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Statistical assessment of motif enrichments in a set of non-random versus randomly sampled gene promotor sequences for multiple species — motif_enrichment_multi_promotor_seqs • metablastr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Statistical assessment of motif enrichments in a set of non-random versus randomly sampled gene promotor sequences for multiple species — motif_enrichment_multi_promotor_seqs"><meta property="og:description" content="Compare the number of motifs in a set of non-random versus randomly sampled gene promotor sequences within a set of subject genomes. 
The resulting values are then used to statistically assess the enrichment of certain motifs in real sequences compared to randomly sampled gene promotor sequences."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metablastr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/metablastr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Statistical assessment of motif enrichments in a set of non-random versus randomly sampled gene promotor sequences for multiple species</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/metablastr/blob/HEAD/R/motif_enrichment_multi_promotor_seqs.R" class="external-link"><code>R/motif_enrichment_multi_promotor_seqs.R</code></a></small>
    <div class="hidden name"><code>motif_enrichment_multi_promotor_seqs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Compare the number of motifs in a set of non-random versus randomly sampled gene promotor sequences within a set of subject genomes. 
The resulting values are then used to statistically assess the enrichment of certain motifs in real sequences compared to randomly sampled gene promotor sequences.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">motif_enrichment_multi_promotor_seqs</span><span class="op">(</span></span>
<span>  <span class="va">blast_tbl</span>,</span>
<span>  <span class="va">subject_genomes</span>,</span>
<span>  <span class="va">annotation_files</span>,</span>
<span>  annotation_format <span class="op">=</span> <span class="st">"gff"</span>,</span>
<span>  test <span class="op">=</span> <span class="st">"fisher"</span>,</span>
<span>  alternative <span class="op">=</span> <span class="st">"two.sided"</span>,</span>
<span>  <span class="va">interval_width</span>,</span>
<span>  <span class="va">motifs</span>,</span>
<span>  max.mismatch <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  min.mismatch <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>blast_tbl</dt>
<dd><p>a blast_table.</p></dd>


<dt>subject_genomes</dt>
<dd><p>a character vector storing the file paths to the subject genomes that shall be used as subject references.</p></dd>


<dt>annotation_files</dt>
<dd><p>a character vector storing the file paths to the subject annotation files in <code>.gff</code> format that match the subject genomes.</p></dd>


<dt>annotation_format</dt>
<dd><p>the annotation format. Options are:</p><ul><li><p><code>annotation_format = "gff"</code></p></li>
</ul></dd>


<dt>test</dt>
<dd><ul><li><p><code>test = "fisher"</code>: Fisher's Exact Test for Count Data (see <code>link[stats]{fisher.test}</code> for details).</p></li>
</ul></dd>


<dt>alternative</dt>
<dd><p>indicates the alternative hypothesis and must be one of <code>"two.sided"</code>, <code>"greater"</code> or <code>"less"</code>. You can specify just the initial letter. Only used in the 2 by 2 case.</p></dd>


<dt>interval_width</dt>
<dd><p>total number of sequences that shall be sampled per subject genome.</p></dd>


<dt>motifs</dt>
<dd><p>a character vector storing (case sensitive) motif sequences for which abundance in the sampled sequences shall be assessed.</p></dd>


<dt>max.mismatch</dt>
<dd><p>maximum number of mismatches that are allowed between the sequence motif and the matching region in the sampled sequence.</p></dd>


<dt>min.mismatch</dt>
<dd><p>minimum number of mismatches that are allowed between the sequence motif and the matching region in the sampled sequence.</p></dd>


<dt>...</dt>
<dd><p>additional arguments passed to <code><a href="motif_compare.html">motif_compare</a></code>.</p></dd>

</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="motif_count.html">motif_count</a></code>, <code><a href="motif_compare_multi_promotor_seqs.html">motif_compare_multi_promotor_seqs</a></code>, <code><a href="motif_compare.html">motif_compare</a></code>, 
<code><a href="motif_enrichment.html">motif_enrichment</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Anna Gogleva.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

